### PR TITLE
Add further considerations for CASE expressions

### PR DESCRIPTION
**Pull request by gem-neo4j**

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1013.surge.sh

Updated pages:
- [Conditional expressions (CASE)](https://neo4j-docs-cypher-1013.surge.sh/cypher-manual/5/queries/case)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1013.surge.sh

Updated pages:
- [Conditional expressions (CASE)](https://neo4j-docs-cypher-1013.surge.sh/cypher-manual/5/queries/case)

**Review by Lojjs — APPROVED** [review]
Good explanation

File: modules/ROOT/pages/queries/case.adoc [hunk 1]
```diff
@@ -323,3 +323,27 @@ RETURN n.name, n.colorCode
...
+----
+WITH "2024-08-05" AS date, "string" AS type
+RETURN CASE type
+    WHEN "string" THEN datetime(date)
+    WHEN "date" THEN datetime({year: date.year, month: date.month, day: date.day})
+    ELSE datetime(date)
+END AS dateTime
+----
+
+[source, error]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/queries/case.adoc]:
```suggestion

.Error message
[source, error]
```

File: modules/ROOT/pages/queries/case.adoc [hunk 2]
```diff
@@ -323,3 +323,27 @@ RETURN n.name, n.colorCode
...
 For more information about using the `SET` clause, see xref::clauses/set.adoc[SET].
+
+== Further considerations
+
+`CASE` result branches are statically checked prior to execution. 
+This means that if a branch is not semantically correct, it will still throw an exception, even if that branch may never be executed during runtime.
+
+In the following example, `date` is statically known to be a `STRING` value, and therefore would fail if treated as a `DATE` value. 
+
+[source, cypher, role=test-fail]
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/queries/case.adoc]:
```suggestion

.Not allowed
[source, cypher, role=test-fail]
```

**Review by JPryce-Aklundh — APPROVED** [review]
Great 👍

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1013.surge.sh

Updated pages:
- [Conditional expressions (CASE)](https://neo4j-docs-cypher-1013.surge.sh/cypher-manual/5/queries/case)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1013.surge.sh

Updated pages:
- [Conditional expressions (CASE)](https://neo4j-docs-cypher-1013.surge.sh/cypher-manual/5/queries/case)

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**